common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -845,14 +845,34 @@ public static int hashCodeAscii(CharSequence bytes) {
      * Create a new {@link Queue} which is safe to use for multiple producers (different threads) and a single
      * consumer (one thread!).
      */
-    public static <T> Queue<T> newMpscQueue(int maxCapacity) {
-        return hasUnsafe() ?
-                new MpscChunkedArrayQueue<T>(MPSC_CHUNK_SIZE,
-                        // Calculate the max capacity which can not be bigger then MAX_ALLOWED_MPSC_CAPACITY.
-                        // This is forced by the MpscChunkedArrayQueue implementation as will try to round it
-                        // up to the next power of two and so will overflow otherwise.
-                        Math.max(Math.min(maxCapacity, MAX_ALLOWED_MPSC_CAPACITY), MIN_MAX_MPSC_CAPACITY), true)
-                : new MpscLinkedAtomicQueue<T>();
+    public static <T> Queue<T> newMpscQueue(final int maxCapacity) {
+        Object maybeQueue = null;
+        if (hasUnsafe()) {
+            maybeQueue = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                @Override
+                public Object run() {
+                    // Calculate the max capacity which can not be bigger then MAX_ALLOWED_MPSC_CAPACITY.
+                    // This is forced by the MpscChunkedArrayQueue implementation as will try to round it
+                    // up to the next power of two and so will overflow otherwise.
+                    final int capacity =
+                            Math.max(Math.min(maxCapacity, MAX_ALLOWED_MPSC_CAPACITY), MIN_MAX_MPSC_CAPACITY);
+                    try {
+                        return new MpscChunkedArrayQueue<T>(MPSC_CHUNK_SIZE, capacity, true);
+                    } catch (SecurityException e) {
+                        return e;
+                    }
+                }
+            });
+        }
+
+        if (maybeQueue instanceof Queue) {
+            logger.debug("org.jctools-core.MpscChunkedArrayQueue: available");
+            @SuppressWarnings("unchecked") Queue<T> queue = (Queue<T>) maybeQueue;
+            return queue;
+        } else {
+            logger.debug("org.jctools-core.MpscChunkedArrayQueue: unavailable", (Exception) maybeQueue);
+            return new MpscLinkedAtomicQueue<T>();
```

Comment:
@jasontedor couldn't we try to create on the static block and so only log once ?